### PR TITLE
paramsd: cache the params with the capnp struct

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Can you write a test for this to ensure we read the cache correctly? There's a similar one for calibrationd you can use as an example.

File: selfdrive/locationd/test/test_paramsd.py [hunk 1]
```diff
@@ -6,7 +6,6 @@
 import cereal.messaging as messaging
 from cereal import car
 from common.params import Params
-from selfdrive.locationd.paramsd import load_params
```
Comment (deanlee) [inline selfdrive/locationd/test/test_paramsd.py]: import here will cause `self.localizer = self.lib.localizer_init()` failed in `selfdrive/locationd/test/test_locationd.py` with the following assertion:

> python3.8: rednose/helpers/ekf_sym.cc:14: EKFS::EKFSym::EKFSym(std::string, Map<EKFS::MatrixXdr>, Map<Eigen::VectorXd>, Map<EKFS::MatrixXdr>, int, int, int, int, int, std::vector<int>, std::vector<int>, std::vector<std::string>, double): Assertion `this->ekf' failed.

**Comment by pd0wm** [comment]
Not a fan of this bytes in JSON. If the message isn't completely statically typed I don't see much of an advantage.

**Comment by deanlee** [comment]
I'm not sure where to call CLEAR_ON_CAR_CHANGE would be more appropriate.

File: selfdrive/common/params.cc [hunk 2]
```diff
@@ -137,7 +137,7 @@ std::unordered_map<std::string, uint32_t> keys = {
     {"LastSystemShutdown", CLEAR_ON_MANAGER_START},
     {"LastUpdateException", PERSISTENT},
     {"LastUpdateTime", PERSISTENT},
-    {"LiveParameters", PERSISTENT},
+    {"LiveParameters", PERSISTENT | CLEAR_ON_CAR_CHANGED},
```
Comment (pd0wm) [inline selfdrive/common/params.cc:140]: You can update `CalibrationParams` as well.

Comment (pd0wm) [reply selfdrive/common/params.cc:140]: never mind, let's leave calibrationd for what it is. It currently also doesn't reset on car change.

**Comment by pd0wm** [comment]
That should probably happen in controlsd when the fingerprint happens, before it writes the carParams.

**Comment by deanlee** [comment]
how about use CarParamsCache and set it to PERSISTENT?

**Pull request closed without merging**